paddle/fluid/pybind/eager.cc

```diff
@@ -155,52 +187,175 @@ void InitEagerTensorWithEagerTensor(EagerTensorObject* self,
   }
 }
 
-void InitEagerTensorWithFrameworkTensor(EagerTensorObject* self,
-                                        const framework::Tensor& src,
-                                        const paddle::platform::Place& place,
-                                        const std::string& name) {
-  self->eager_tensor.set_name(name);
-  if (place == src.place()) {
-    std::shared_ptr<pten::DenseTensor> dense_tensor =
-        std::make_shared<pten::DenseTensor>(
-            pten::make_intrusive<paddle::experimental::SharedStorage>(place),
-            pten::DenseTensorMeta(pten::TransToPtenDataType(src.type()),
-                                  src.dims()));
-    paddle::experimental::ReMakePtenDenseTensor(src, dense_tensor.get());
-    self->eager_tensor.set_impl(dense_tensor);
-    VLOG(4) << "Same place, do ShareDataWith";
+// initialize EagerTensor by PyArray (mix args and kwargs) automatically
+void AutoInitEagerTensorByPyArray(
+    EagerTensorObject* py_tensor_ptr,
+    std::unordered_map<std::string, PyObject*> kws_map, PyObject* args,
+    bool flag_kwargs, Py_ssize_t args_num) {
+  // key_word lists: [value, place, persistable, zero_copy, name, stop_gradient]
+  std::unordered_map<std::string, Py_ssize_t> kw_order_map{
+      {"value", 1},     {"place", 2}, {"persistable", 3},
+      {"zero_copy", 4}, {"name", 5},  {"stop_gradient", 6}};
+
+  py::object numpy_value = py::object();
+  paddle::platform::Place place =
+      egr::Controller::Instance().GetExpectedPlace();
+  bool persistable = false;
+  bool zero_copy = false;
+  std::string act_name = "";
+  bool stop_gradient = true;
+
+  if (kw_order_map["value"] <= args_num) {
```

Comment:
Will this if statement work if we have kwargs and args?

Reply:
This if statement just work for args, but the else statement will work for kwargs.